src/js/stores/AppFormValidators.js

```diff
@@ -7,12 +8,27 @@ const AppFormValidators = {
     !Util.isEmptyString(value) && value.toString().match(/^[0-9\.]+$/),
   disk: (value) =>
     !Util.isEmptyString(value) && value.toString().match(/^[0-9\.]+$/),
+  constraints: (constraints) => { return constraints
+    .split(",")
+    .map((constraint) => constraint.split(":").map((value) => value.trim()))
+    .every((p) => {
+      if (p.length < 2 || p.length > 3) {
+        return false;
+      }
+      return ValidConstraints.indexOf(p[1].toLowerCase()) !== -1;
+    });
+  },
   env: (obj) =>
     !(Util.isEmptyString(obj.key) && !Util.isEmptyString(obj.value)),
+  executor: (str) => Util.isString(str) &&
+    (new RegExp("^(|\\/\\/cmd|\\/?[^\\/]+(\\/[^\\/]+)*)$"))
```

Comment:
Jokes aside, I wonder if we really need to care about the executor value. IMHO Marathon itself should do this kind of validation.